src/main/java/com/zufar/onlinestore/security/configuration/SpringSecurityConfiguration.java

```diff
@@ -60,12 +60,12 @@ public SecurityFilterChain securityFilterChain(final HttpSecurity httpSecurity,
     @Bean
     public UserDetailsService userDetailsService(final UserRepository userRepository,
                                                  final PasswordEncoder passwordEncoder) {
-        return username -> {
-            UserEntity user = userRepository.findUserByUsername(username);
-            if (user == null) {
-                log.warn("Failed to get the user with the username = {}.", username);
-                throw new BadCredentialsException("Bad credentials");
-            }
+        return email -> {
+            UserEntity user = userRepository.findByEmail(email)
+                    .orElseThrow(() -> {
+                        log.warn("Failed to get the user with the email = {}.", email);
+                        return new BadCredentialsException("Bad credentials");
```

Comment:
Maybe we should follow our default pattern to provide a message in parent constructor call in the class?